### ISSUE TITLE
Unit Tests Setup

### ISSUE THREAD

**Issue by francescobm** [nuxt/nuxt.js#1709]
I see that Nuxt.js comes already with e2e tests setup. 
However, for my project I would like also to implement Karma Mocha and Chai to write unit tests for my components and for the Vuex store.

Someone has any suggestion on how to go about it ?
Is there any plan to integrate unit tests in the nuxt.config.js in the future ?

Thanks for the wonderful work by the way, Nuxt is fantastic!

<div align="right"><sub><em>This question is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c1529">#c1529</a>)</em></sub></div>

**Comment by francescobm** [comment]
Closing as I actually wrongly assumed from the docs that AVA is for e2e only. 
Instead I can use it to write unit tests too.

**Comment by uptownhr** [comment]
I would love to see some examples as well

**Comment by alidcastano** [comment]
Not exactly karma-mocha-chai, but here's how I configured jest and puppeteer for nuxt, maybe it'll help: https://github.com/studbits/nuxt-jest-puppeteer

**Comment by francescobm** [comment]
Hi @alidcastano, awesome! Thanks for pointing this out, I'll try this

**Comment by PMK** [comment]
The official VueJS test framework [vue-test-utils](https://github.com/vuejs/vue-test-utils) is using [mocha-webpack](https://github.com/zinserjan/mocha-webpack), where you can specify a webpack-config file. I'm now trying to get Mocha working in Nuxt using vue-test-utils... Does Nuxt expose its generated Webpack config?

```
mocha-webpack --webpack-config mocha-webpack.config.js --require test/_helpers/mocha-setup.js test/unit/**/*.spec.js
```

Should I use `dllWebpackConfig` from the Nuxt core? Or do I need to use a hook on the Builder class? I'm kinda lost...

**Comment by PMK** [comment]
@tappoz Sure, check this Gist: https://gist.github.com/PMK/27063172a2d0e6d5b1aa2a1ba3308b79

**Comment by uptownhr** [comment]
@pi0 @Atinux any answer to @PMK question in regards to getting an export of nuxt's webpack.config? Maybe expose it as part of the /dist folder as it builds?

**Comment by Atinux** [comment]
Not right now, but I plan to be able to get programmatically the webpack config so it could be plugged into any testing tool that need the Webpack config.

**Comment by uptownhr** [comment]
looking forward to it!

**Comment by Chathula** [comment]
+1 for this

**Comment by manniL** [comment]
Resolved by #4163 and #4116. For e2e: https://github.com/melrefaie/nuxt-cypress
